src/cascadia/PublicTerminalCore/HwndTerminal.cpp

```diff
@@ -0,0 +1,413 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "HwndTerminal.hpp"
+#include <DefaultSettings.h>
+#include "../../renderer/base/Renderer.hpp"
+#include "../../renderer/dx/DxRenderer.hpp"
+#include "../../cascadia/TerminalCore/Terminal.hpp"
+#include "../../types/viewport.cpp"
+#include "../../types/inc/GlyphWidth.hpp"
+
+using namespace ::Microsoft::Terminal::Core;
+
+static LPCWSTR term_window_class = L"HwndTerminalClass";
+
+LRESULT CALLBACK HwndTerminalWndProc(
+    HWND hwnd,
+    UINT uMsg,
+    WPARAM wParam,
+    LPARAM lParam) noexcept
+{
+    return DefWindowProc(hwnd, uMsg, wParam, lParam);
+}
+
+bool RegisterTermClass(HINSTANCE hInstance) noexcept
+{
+    WNDCLASSW wc;
+    if (GetClassInfo(hInstance, term_window_class, &wc))

```

Comment:
Should this be `GetClassInfoW`?

Reply:
I thought that the unqualified functions were defined to the wide string variants based on the fact that we were compiling with unicode support. Should I use the wide string variants explicitly in this codebase?

Reply:
They are, generally, so this is just a nit. I think we're trying to remain "clean" of any defs that could accidentally switch themselves over to ANSI, but I don't know how much we care.

Reply:
Given you declared WNDCLASSW, use GetClassInfoW.

If UNICODE was undefined (which it would never be... really), then it would fail to compile because we'd have an A/W mismatch.

Reply:
To be fair, I don't super care. I just bias toward going all W because its 2019 and A is icky.